qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/utils/CDIUtils.java

```diff
@@ -63,4 +73,35 @@ public static String getSimpleName(String javaElementName, String annotationName
 		}
 		return javaElementName;
 	}
+
+	public static boolean isValidBean(IJavaElement javaElement) {
+		try {
+			return (((IType) javaElement).getDeclaringType() == null
+					&& !Modifier.isAbstract(javaElement.getClass().getModifiers())

```

Comment:
This part is still wrong. Look at what `Modifier` & `javaElement.getClass()` refer to. They're from `java.lang.reflect`, so they won't do what you want.

Look inside JDT-LS for `isAbstract` . You'll find some references to methods that do it for AST Nodes (which aren't useful here). You'll find other references that do it for IBindings (also not useful here), but there will be some references to how to do it for IJavaElement (eg. IMethod, IType).